From af8d874f9fb7389b3c6a3b6cba3d89467de85ee5 Mon Sep 17 00:00:00 2001
From: Finley Xiao <finley.xiao@rock-chips.com>
Date: Wed, 1 Mar 2017 18:00:48 +0800
Subject: [PATCH] PM / devfreq: rk3399_dmc: initialize min_freq and max_freq

In order to get correct results from sysfs, e.g.:
cat /sys/class/devfreq/dmc/min_freq
cat /sys/class/devfreq/dmc/max_freq

Change-Id: Id5921fdbacd0977c0b5378ccf0de068f0195b557
Signed-off-by: Finley Xiao <finley.xiao@rock-chips.com>
---
 drivers/devfreq/rk3399_dmc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/devfreq/rk3399_dmc.c b/drivers/devfreq/rk3399_dmc.c
index 98312f9b0d62..fc6ed4ddae1c 100644
--- a/drivers/devfreq/rk3399_dmc.c
+++ b/drivers/devfreq/rk3399_dmc.c
@@ -467,6 +467,10 @@ static int rk3399_dmcfreq_probe(struct platform_device *pdev)
 		return PTR_ERR(data->devfreq);
 	devm_devfreq_register_opp_notifier(dev, data->devfreq);
 
+	data->devfreq->min_freq = devp->freq_table[0];
+	data->devfreq->max_freq =
+		devp->freq_table[devp->max_state ? devp->max_state - 1 : 0];
+
 	data->dev = dev;
 	platform_set_drvdata(pdev, data);
 
-- 
2.35.3

